query-service-impl/src/main/java/org/hypertrace/core/query/service/ExecutionContext.java

```diff
@@ -219,12 +225,12 @@ private void extractColumns(List<String> columns, Expression expression) {
 
   private boolean isMatchingFilter(Filter filter, String column, Collection<Operator> operators) {
 
-    // if not leaf filter, then return false
-    if (!filter.hasLhs()) {
+    Optional<String> logicalColumnName = getLogicalColumnName(filter.getLhs());
+    if (logicalColumnName.isEmpty()) {
       return false;
     }
 
-    return column.equals(getLogicalColumnName(filter.getLhs()))
+    return column.equals(logicalColumnName.get())
```

Comment:
In general, `Optional.get()` is an antipattern, it shouldn't even be part of the API in my opinion. If you want to work imperatively, you should use either `orElseThrow()` (which is identical in behavior to `.get()`, but conveys the behavior better, since `get()` will throw too), or `orElse[Get]()` to provide a default. Other options include functional style:
```java
return getLogicalColumnName(filter.getLhs())
  .map(column::equals)
  .orElse(false) && ...
```
or comparing optionals directly rather than unwrapping to do the comparison:
```java
return Optional.of(column).equals(getLogicalColumnName(filter.getLhs())) && ...
```

The benefit of both of these examples is they prevent logic separation. In the current code, line 233 relies on an earlier check (line 229) to behave correctly. This is a fairly trivial case, but anywhere we can reduce logic separation, we're reducing the chance of bugs when a later maintainer comes in and changes one place without realizing another relies on it.

Reply:
This is a nit in this location btw, but trying to give more context why the optionals are actually more maintainable than throwing directly in `getLogicalColumnName`.

Reply:
resolved